Lib/asyncio/staggered.py

```diff
@@ -116,8 +119,10 @@ async def run_one_coro(previous_failed) -> None:
                 if i != this_index:
                     t.cancel()
 
-    first_task = loop.create_task(run_one_coro(None))
+    ok_to_start = locks.Event()
+    first_task = loop.create_task(run_one_coro(ok_to_start, None))
     running_tasks.append(first_task)
+    ok_to_start.set()
```

Comment:
For the first one, should the event be preset so it doesn't negate the benefits of an eager task factory?  

Or maybe make the event optional for the first one since we want it to start right away?

Reply:
it's not terribly important to maintain the benefits of an eager task factory here as it's supposed to be used for happy_eyeballs over the internet where there is always going to be some delay

Reply:
I was only thinking about the first one since there is a chance the first ip could connect right away (not sure if it can happen synchronously or not)

In aiohttp we don't know if the host is an internet or lan or localhost so everything ends up going through this path.

Reply:
Even connecting to a localhost socket raises BlockingIOError and asyncio needs to wait for it to be writable